### PR TITLE
WIP: add model error analysis

### PR DESCRIPTION
**Pull request by benisraeldan**
find top features that contribute to model error.

### CODE REVIEW

File: deepchecks/checks/performance/model_error_analysis.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+                grouped_category = scored_feature.groupby(feat)
+                display.append(partial(display_categorical, display_data, feature, grouped_category))
+            else:
+                display.append(partial(display_numeric, display_data, feature))
+
+        value = None
+        headnote = """<span>
+            The following graphs show the top features that contribute to error and their values compared to the error.
+            </br>
+            Categorical features are represented as a violin graph. (x axis: category, y axis: error)
```
Comment (noamzbr) [inline deepchecks/checks/performance/model_error_analysis.py:155]: Not sure this is needed if graph is improved.

File: deepchecks/checks/performance/model_error_analysis.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+            if feature in cat_features:
+                feat = dataset.data[feature]
+                scored_feature = pd.DataFrame(feat).assign(score=score)
+                grouped_category = scored_feature.groupby(feat)
+                display.append(partial(display_categorical, display_data, feature, grouped_category))
+            else:
+                display.append(partial(display_numeric, display_data, feature))
+
+        value = None
+        headnote = """<span>
```
Comment (noamzbr) [inline deepchecks/checks/performance/model_error_analysis.py:152]: This explanation should talk way more on what we're actually doing here, it's non trivial

File: deepchecks/checks/performance/model_error_analysis.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    def run(self, dataset: Dataset, model) -> CheckResult:
+        """Run check.
+
+        Args:
+            dataset (Dataset): a Dataset object.
+            model (BaseEstimator): A scikit-learn-compatible fitted estimator instance.
+        """
+        # Validations
+        Dataset.validate_dataset(dataset)
+        dataset.validate_label()
```
Comment (noamzbr) [inline deepchecks/checks/performance/model_error_analysis.py:69]: Use only sample of data

File: deepchecks/checks/performance/model_error_analysis.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+                ('cat', categorical_transformer, cat_features),
+            ]
+        )
+
+        error_model = Pipeline(steps=[
+            ('preprocessing', preprocessor),
+            ('model', RandomForestRegressor(max_depth=4, n_jobs=-1, random_state=self.random_seed))
+        ])
+
+        error_model.fit(dataset.features_columns, y=score)
```
Comment (noamzbr) [inline deepchecks/checks/performance/model_error_analysis.py:101]: fit on train split of this dataset

File: deepchecks/checks/performance/model_error_analysis.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+        error_fi = calculate_feature_importance(error_model, dataset)
```
Comment (noamzbr) [inline deepchecks/checks/performance/model_error_analysis.py:103]: evaluate fi on test split

**Review by noamzbr — CHANGES_REQUESTED** [review]

File: deepchecks/checks/performance/model_error_analysis.py [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+        error_fi.sort_values(ascending=False, inplace=True)
+
+        display_data = dataset.data.assign(score=score)
```
[the remark this answers is not in the document]
Comment (benisraeldan) [reply deepchecks/checks/performance/model_error_analysis.py:95]: saves the data to display, which in this case is the dataset and score, it's only needed for the callbacks, will be removed for plotly

**Pull request closed without merging**